File: Question_LeetcodeAndGeeks/Greedy/medium/Q011Leetcode732_my-calendar-iii/Q011Leetcode732.java
import java.util.TreeMap;
import java.util.Map;

class MyCalendarThree {
    TreeMap < Integer, Integer > map = new TreeMap < > ();
    public MyCalendarThree() {
         map.clear();
    }
    
    public int book(int start, int end) {
         map.put(start, map.getOrDefault(start, 0) + 1);
            map.put(end, map.getOrDefault(end, 0) - 1);

            int count = 0,res=0;
            for (Map.Entry < Integer, Integer > key: map.entrySet()) {
                count += key.getValue();
                res=Math.max(res,count);
            }
            return res;
    }
}

File: Question_LeetcodeAndGeeks/Greedy/medium/Q011Leetcode732_my-calendar-iii/Q011Leetcode732.cpp
// #include <bits/stdc++.h>
#include <iostream>
#include <vector>
#include <string>

#define vi vector<int>
#define vii vector<vi>

using namespace std;

static const auto magic = []() {
    std::ios::sync_with_stdio(false);
    std::cin.tie(nullptr);
    std::cout.tie(nullptr);
    return nullptr;
}();

class MyCalendarThree {
public:
    map<int,int> bst;
     MyCalendarThree() {
        bst.clear();
    }
    
     int book(int start, int end) {
        bst[start]++;
        bst[end]--;
        
        int count=0,res=0;
        for(auto ele: bst){
            count += ele.second;
            res=max(count,res);
        }
        return res;
    }
};


